core/src/test/java/com/linecorp/armeria/server/VirtualHostAnnotatedServiceBindingBuilderTest.java

```diff
@@ -0,0 +1,117 @@
+
+    private static final ExceptionHandlerFunction handlerFunction = (ctx, req, cause) -> HttpResponse.of(501);
+    private static final String TEST_HOST = "foo.com";
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.virtualHost(TEST_HOST)
+              .annotatedService()
+              .pathPrefix("/foo")
+              .exceptionHandler(handlerFunction)
+              .build(new TestService());
+        }
+    };
+
+    private static CloseableHttpResponse get(String path) throws Exception {
+        try (CloseableHttpClient hc = HttpClients.createMinimal()) {
+            final HttpGet httpGet = new HttpGet(server.httpUri(path));
+            httpGet.setHeader(HttpHeaderNames.HOST.toString(), TEST_HOST);
+            try (CloseableHttpResponse response = hc.execute(httpGet)) {
+                return response;
+            }
+        }
+    }
+
+    @Test
+    void testAllConfigsAreSet() {
+        final boolean verboseResponse = true;
+        final boolean shutdownOnStop = true;
+        final long maxRequestLength = 2 * 1024;
+        final AccessLogWriter accessLogWriter = AccessLogWriter.common();
+        final Duration requestTimeoutDuration = Duration.ofMillis(1000);
+        final ContentPreviewerFactory factory = (ctx, headers) -> ContentPreviewer.ofText(1024);
+
+        final VirtualHost virtualHost = new VirtualHostBuilder(Server.builder(), false)
+                .annotatedService()
+                .requestTimeout(requestTimeoutDuration)
+                .maxRequestLength(maxRequestLength)
+                .exceptionHandler((ctx, request, cause) -> HttpResponse.of(400))
+                .pathPrefix("/foo")
+                .accessLogWriter(accessLogWriter, shutdownOnStop)
+                .contentPreviewerFactory(factory)
+                .verboseResponses(verboseResponse)
+                .build(new TestService())
+                .build();
+
+        assertThat(virtualHost.serviceConfigs()).hasSize(1);
+        final ServiceConfig serviceConfig = virtualHost.serviceConfigs().get(0);
+        assertThat(serviceConfig.route().paths()).allMatch("/foo/bar"::equals);
+        assertThat(serviceConfig.requestTimeoutMillis()).isEqualTo(requestTimeoutDuration.toMillis());
+        assertThat(serviceConfig.maxRequestLength()).isEqualTo(maxRequestLength);
+        assertThat(serviceConfig.accessLogWriter()).isEqualTo(accessLogWriter);
+        assertThat(serviceConfig.shutdownAccessLogWriterOnStop()).isTrue();
+        assertThat(serviceConfig.requestContentPreviewerFactory()).isEqualTo(factory);
+        assertThat(serviceConfig.verboseResponses()).isTrue();
+    }
+
+    @Test
+    @Ignore
+    void testServiceDecoration_shouldCatchException() throws Exception {
+        try (CloseableHttpClient hc = HttpClients.createMinimal()) {
```

Comment:
I tried using Armeria's http client and it seems that it is not respecting Host header values. IDK if this is intended behaviour, but I feel a bit weird on why that would be an intended behaviour though. So had to use Apache's http client. 

My Armeria's http client code is as follows:

```
final HttpClient httpClient = new HttpClientBuilder(server.httpUri("/"))
                .decorator((delegate, ctx, req) -> {
                    ctx.addAdditionalRequestHeader(HttpHeaderNames.METHOD, "GET");
                    ctx.addAdditionalRequestHeader(HttpHeaderNames.HOST, TEST_HOST);
                    return delegate.execute(ctx, req);
                })
                .build();
        return httpClient.get("/foo/bar").aggregate().join();
```
And I get 404. Let me know if I should create an issue for this.

Reply:
To use the Armeria client with a custom host I think you need to initialize it with an `Endpoint`

`Endpoint.of(host, port).withIpAddr(127.0.0.1);`

Reply:
I think you need to use .AUTHORITY rather than .HOST. (because it's HTTP/2)

Reply:
.. and therr's no beed to set .GET.

Reply:
Thank you @anuraaga @trustin for the advice 🙏 . I was able to use Armeria http client.